index.html

```diff
@@ -548,73 +562,94 @@ <h2>
         MAY support content codings defined in previous versions of the draft for compatibility
         reasons.
       </p>
+      <h3>
+        `subscribe()` method
+      </h3>
       <p>
-        The <dfn>subscribe</dfn> method when invoked MUST run the following steps:
+        The <dfn>subscribe()</dfn> method when invoked MUST run the following steps:
       </p>
-      <ol>
-        <li>Let |promise| be <a>a new promise</a>.
+      <ol class="algorithm">
+        <li>If the |options| argument has a {{PushSubscriptionOptionsInit/userVisibleOnly}} value
+        set to `false` and the user agent requires it to be `true`, return [=a promise rejected
+        with=] an {{"NotAllowedError"}} {{DOMException}}.
         </li>
-        <li>Return |promise| and continue the following steps asynchronously.
+        <li>If the |options| argument does not include a non-null value for the
+        {{PushSubscriptionOptionsInit/applicationServerKey}} member, and the <a>push service</a>
+        requires one to be given, return [=a promise rejected with=] a {{"NotSupportedError"}}
+        {{DOMException}}.
+        </li>
+        <li>Let |promise| be [=a new promise=].
+        </li>
+        <li>Let |global:Window| be [=this=]' [=relevant global object=].
```

Comment:
The relevant global object within a service worker isn't a window, so this definition may be invalid. None of the uses of `global` seem to require it to be a Window though, the worker global (in my reading) also has a task queue.

Reply:
Ah, yeah. good point. I'll remove the type.